### PR TITLE
Reword the cautionary line about "specification" in Monillo

### PR DESCRIPTION
**Pull request by pradyunsg**
It seems to be an overtly specific reminder from back-when-we-were-looking-around-to-understand-what-to-do.

### CODE REVIEW

**Comment by uranusjr** [comment]
IMO the line is still worthwhile, since [there are people using the term in Python packaging](https://discuss.python.org/t/structured-exchangeable-lock-file-format-requirements-txt-2-0/876/96?u=uranusjr). I wouldn’t object dropping the Monillo mention, however.

**Comment by pradyunsg** [comment]
HEADS UP: we should update some documentation string in providers.py as well. Hold off on merging please!

**Comment by pradyunsg** [comment]
I've skipped one instance in `provider.py` since I covered it in #53 (https://github.com/sarugaku/resolvelib/pull/53/files#diff-db65315aa1492e1d778ccbe3921352faR10)

File: src/resolvelib/providers.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def identify(self, dependency):
         raise NotImplementedError
 
     def get_preference(self, resolution, candidates, information):
-        """Produce a sort key for given specification based on preference.
+        """Produce a sort key for given candidate based on preference.
```
Comment (uranusjr) [inline src/resolvelib/providers.py:16]: This should be *requirement*, not candidate. (same below)

Comment (pradyunsg) [reply src/resolvelib/providers.py:16]: Ahahahah

**Pull request merged**